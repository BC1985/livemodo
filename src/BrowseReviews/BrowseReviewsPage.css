/* Global settings */

.reviews-container {
  color: whitesmoke;
  padding: 5% 10%;
}
.band-name {
  text-align: left;
}

.tagline {
  color: gray;
  font-style: italic;
}
.review {
  margin-bottom: 10%;
}
.review > * {
  margin: 6px;
}
.review {
  margin-bottom: 20%;
}
/* ----------- iPhone 5, 5S, 5C and 5SE ----------- */

/* Portrait */
@media only screen and (min-device-width: 320px) and (max-device-width: 568px) and (-webkit-min-device-pixel-ratio: 2) and (orientation: portrait) {
  .reviews-container {
    color: whitesmoke;
  }

  .username {
    margin: 10px;
    font-size: 1.5em;
  }

  .user-thumbnail {
    font-size: 50px;
  }

  .push {
    height: 50px;
  }
  .review > * {
    margin: 6px;
  }
  .review {
    margin-bottom: 20%;
  }
}

/* Landscape */
@media only screen and (min-device-width: 320px) and (max-device-width: 568px) and (-webkit-min-device-pixel-ratio: 2) and (orientation: landscape) {
  .user-thumbnail {
    font-size: 50px;
  }
  .username {
    margin: 10px;
    font-size: 1.5em;
  }
  .review > * {
    margin: 6px;
  }
  .review {
    margin-bottom: 20%;
  }
}

/* ----------- iPhone 6, 6S, 7 and 8 ----------- */

/* Portrait */
@media only screen and (min-device-width: 375px) and (max-device-width: 667px) and (-webkit-min-device-pixel-ratio: 2) and (orientation: portrait) {
  .user-thumbnail {
    font-size: 50px;
  }
  .review > * {
    margin: 6px;
  }
  .review {
    margin-bottom: 20%;
  }
}

/* Landscape */
@media only screen and (min-device-width: 375px) and (max-device-width: 667px) and (-webkit-min-device-pixel-ratio: 2) and (orientation: landscape) {
  .reviews-container {
    /* font-size: 1.1em; */
  }
  .user-thumbnail {
    font-size: 50px;
  }
  .username {
    margin: 10px;
    font-size: 1.5em;
  }
  .review > * {
    margin: 6px;
  }
  .review {
    margin-bottom: 20%;
  }
}

/* ----------- iPhone 6+, 7+ and 8+ ----------- */

/* Portrait */
@media only screen and (min-device-width: 414px) and (max-device-width: 736px) and (-webkit-min-device-pixel-ratio: 3) and (orientation: portrait) {
  .reviews-container {
    /* font-size: 1.1em; */
  }
  .user-thumbnail {
    font-size: 50px;
  }
  .username {
    margin: 10px;
    font-size: 1.5em;
  }
  .review > * {
    margin: 6px;
  }
  .review {
    margin-bottom: 20%;
  }
}

/* Landscape */
@media only screen and (min-device-width: 414px) and (max-device-width: 736px) and (-webkit-min-device-pixel-ratio: 3) and (orientation: landscape) {
  .reviews-container {
    /* font-size: 1.1em; */
    padding: 5% 20% 5% 10%;
  }
  .user-thumbnail {
    font-size: 50px;
  }
  .username {
    margin: 10px;
    font-size: 1.5em;
  }
  .review > * {
    margin: 6px;
  }
  .review {
    margin-bottom: 20%;
  }
}

/* ----------- iPhone X ----------- */

/* Portrait */
@media only screen and (min-device-width: 375px) and (max-device-width: 812px) and (-webkit-min-device-pixel-ratio: 3) and (orientation: portrait) {
  .reviews-container {
    padding: 5% 15%;
  }
  .user-thumbnail {
    font-size: 50px;
  }
  .username {
    margin: 10px;
    font-size: 1.5em;
  }
}

/* Landscape */
@media only screen and (min-device-width: 375px) and (max-device-width: 812px) and (-webkit-min-device-pixel-ratio: 3) and (orientation: landscape) {
  .reviews-container {
    padding: 5% 20% 15% 10%;
    /* font-size: 1.2em; */
  }
  .user-thumbnail {
    font-size: 50px;
  }
  .username {
    margin: 10px;
    font-size: 1.5em;
  }
  .review > * {
    margin: 6px;
  }
  .review {
    margin-bottom: 20%;
  }
}
/* ----------- iPad 3, 4 and Pro 9.7" ----------- */

/* Portrait */
@media only screen and (min-device-width: 768px) and (max-device-width: 1024px) and (orientation: portrait) and (-webkit-min-device-pixel-ratio: 2) {
  .reviews-container {
    padding: 5% 15%;
    /* font-size: 1.2em; */
  }
  .user-thumbnail {
    font-size: 50px;
  }
  .username {
    margin: 10px;
    font-size: 1.5em;
  }
  .review > * {
    margin: 6px;
  }
  .review {
    margin-bottom: 10%;
  }
}

/* Landscape */
@media only screen and (min-device-width: 768px) and (max-device-width: 1024px) and (orientation: landscape) and (-webkit-min-device-pixel-ratio: 2) {
  .reviews-container {
    padding: 5% 25% 5% 10%;
    /* font-size: 1.2em; */
  }
  .user-thumbnail {
    font-size: 50px;
  }
  .username {
    margin: 10px;
    font-size: 1.5em;
  }
  .review > * {
    margin: 6px;
  }
  .review {
    margin-bottom: 10%;
  }
}

/* ----------- Non-Retina Screens ----------- */
@media screen and (min-device-width: 1200px) and (max-device-width: 1600px) and (-webkit-min-device-pixel-ratio: 1) {
  .reviews-container {
    /* font-size: 1.1em; */
  }
  .user-thumbnail {
    font-size: 50px;
  }
  .username {
    margin: 10px;
    font-size: 1.5em;
  }
  .review > * {
    margin: 6px;
  }
  .review {
    margin-bottom: 10%;
  }
}

/* ----------- Retina Screens ----------- */
@media screen and (min-device-width: 1200px) and (max-device-width: 1600px) and (-webkit-min-device-pixel-ratio: 2) and (min-resolution: 192dpi) {
  .reviews-container {
    /* font-size: 1.1em; */
  }
  .user-thumbnail {
    font-size: 50px;
  }
  .username {
    margin: 10px;
    font-size: 1.5em;
  }
  .review > * {
    margin: 6px;
  }
  .review {
    margin-bottom: 10%;
  }
}

@media screen and (min-device-width: 1600px) {
  .reviews-container {
    padding: 5% 25% 5% 10%;
    /* line-height: 20px;*/
    font-size: 0.9em;
  }
  .user-thumbnail {
    font-size: 3em;
  }
  .username {
    margin: 10px;
    font-size: 1.3em;
  }
  .review > * {
    margin: 6px;
  }
  .review {
    margin-bottom: 10%;
  }
}
