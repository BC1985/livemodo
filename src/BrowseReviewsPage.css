.tagline {
  color: gray;
  font-style: italic;
}
.band-search {
  margin-top: 60px;
  background-color: lightgray;
  padding: 20px;
}

.browse-reviews-container {
  padding: 40px;
  text-align: left;
}
.user-name {
  font-weight: bold;
}
.user-status {
  font-style: italic;
  font-weight: normal;
  color: gray;
}
.review {
  margin-bottom: 100px;
}
