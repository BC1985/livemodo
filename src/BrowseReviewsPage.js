import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faStar } from "@fortawesome/free-solid-svg-icons";
import { faPortrait } from "@fortawesome/free-solid-svg-icons";

function BrowseReviewsPage() {
  const star = <FontAwesomeIcon icon={faStar} />;
  const userThumbnail = <FontAwesomeIcon icon={faPortrait} />;

  return (
    <div className="browse-reviews-container">
      <div className="reviews-section">
        <h2>Browse latest reviews</h2>
        <div className="review-container">
          <h3 className="band-name">Sonic Tooth</h3>
          <p>Venue: The Globe Theatre</p>
          <p>Date:12/3/2003</p>
          {star}
          {star}
          {star}
          {star}
          <h4 className="tagline">
            This is a user tagline lorem ipsum dolor sit amet
          </h4>

          <p className="user-name">
            <span className="user-thumbnail">{userThumbnail}</span> Username{" "}
            <span className="user-status">(avid concertgoer)</span>
          </p>
          <div className="review">
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris a
              orci tincidunt, rhoncus ipsum in, tincidunt est. Fusce aliquam
              quam nec vulputate varius. In hac habitasse platea dictumst.
              Vivamus vulputate rhoncus ipsum sollicitudin maximus. Pellentesque
              id risus efficitur, egestas odio sit amet, ege
            </p>
          </div>
        </div>
        <div className="review-container">
          <h3 className="band-name">The Pointy Brothers</h3>
          <p className="venue-name">Venue: Bob's Crab Shack</p>
          <p>Date:05/11/2012</p>

          {star}
          {star}
          <h4 className="tagline">
            This is a user tagline lorem ipsum dolor sit amet
          </h4>
          <p className="user-name">
            <span className="user-thumbnail">{userThumbnail}</span> Username{" "}
            <span className="user-status">(Homebody)</span>
          </p>
          <div className="review">
            <p>
              magnis dis parturient montes, nascetur ridiculus mus. Quisque in
              lorem metus. Aenean dolor massa, varius vitae rutrum sed, congue
              at urna. Proin et ex at velit feugiat finibus at eget nunc.
              Suspendisse maximus, lacus eget euismod vulputate, lacus mi
              pharetra ante, sed porta lacus leo ac orci
            </p>
          </div>
        </div>
        <div className="review-container">
          <h3 className="band-name">Veruca's Midnight Spacemen</h3>
          <p className="venue-name">Venue: Pompeii ruins</p>
          <p>Date:04/20/2004</p>

          {star}
          {star}
          {star}
          {star}
          {star}
          <h4 className="tagline">
            This is a user tagline lorem ipsum dolor sit amet
          </h4>
          <p className="user-name">
            <span className="user-thumbnail">{userThumbnail}</span> Username{" "}
            <span className="user-status">(avid concertgoer)</span>
          </p>
          <div className="review">
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris a
              orci tincidunt, rhoncus ipsum in, tincidunt est. Fusce aliquam
              quam nec vulputate varius. In hac habitasse platea dictumst.
              Vivamus vulputate rhoncus ipsum sollicitudin maximus. Pellentesque
              id risus efficitur, egestas odio sit amet, ege
            </p>
          </div>
        </div>
      </div>
      <div className="push" />
    </div>
  );
}

export default BrowseReviewsPage;
