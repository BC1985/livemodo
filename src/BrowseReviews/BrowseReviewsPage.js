import React, { Component } from "react";
import "./BrowseReviewsPage.css";
import Review from "../Review/Review";
import config from "../config";

class BrowseReviewsPage extends Component {
  constructor() {
    super();
    this.state = {
      reviews: []
    };
  }
  componentDidMount() {
    this.fetchReviews();
  }
  fetchReviews = async () => {
    const data = await fetch(`${config.API_ENDPOINT}/reviews`);
    const reviews = await data.json();
    this.setState({
      reviews
    });
  };
  render() {
    const { reviews } = this.state;
    const reviewComponents = reviews.map(review => (
      <Review
        key={review.id}
        band_name={review.band_name}
        show_date={review.show_date}
        tagline={review.tagline}
        posted={review.posted}
        venue={review.venue}
        content={review.content}
        username={review.username}
      />
    ));
    return (
      <div className="reviews-container">
        <div className="review">{reviewComponents}</div>
      </div>
    );
  }
}
export default BrowseReviewsPage;
