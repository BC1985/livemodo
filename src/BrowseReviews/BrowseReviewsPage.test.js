import React from "react";
import ReactDOM from "react-dom";
import BrowseReviewsPage from "../BrowseReviews/BrowseReviewsPage";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<BrowseReviewsPage />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer.create(<BrowseReviewsPage />).toJSON();
  expect(tree).toMatchSnapshot();
});
