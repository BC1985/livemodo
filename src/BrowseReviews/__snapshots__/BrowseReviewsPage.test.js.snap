// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="reviews-container"
>
  <div
    className="spinner-3"
  />
  <main
    role="main"
  >
    <div
      className="review"
    />
  </main>
</div>
`;
